core/src/main/scala/dagr/core/tasksystem/Task.scala

```diff
@@ -204,6 +362,21 @@ trait Task extends Dependable {
   def withName(name: String) : this.type = { this.name = name; this }
 
   /** Get the list of tasks to execute.
+    *
+    * All tasks, multi-tasks, workflows, and other task-like-entities should the [[getTasks]] method.
+    *
+    * All executors should call this method to get the list of tasks to execute.
+    *
+    * @return the list of tasks of to run.
+    */
+  final def make(): Traversable[_ <: Task] = {
+    val tasks = getTasks.toSeq
+    tasks.foreach(_._executor = this._executor)
+    this._executor.foreach(_.register(this, tasks:_*))
```

Comment:
A callback to all things that want to know when a task is first seen/registered.